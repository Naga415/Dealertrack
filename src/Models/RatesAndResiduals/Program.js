'use strict';
export default class Program {
  constructor() {
    this.id = null;
    this.type = null;
    this.lender = null;
    this.rebates = null;
    this.rateMoneyFactor = null;
    this.term = null;
    this.acquisitionFee = null;
    this.residual = null;
    this.milesPerYear = null;
    this.payment = null;
    this.frontGross = null;
    this.backGross = null;
    this.expirationDate = null;
    this.maxAdvance = null;
    this.amountFinanced = null;
  }
}
