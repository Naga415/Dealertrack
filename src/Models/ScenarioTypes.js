export const FINANCE = "Finance";
export const LEASE = "Lease";
export const CASH = "Cash";
export const BALLOON = "Balloon";
