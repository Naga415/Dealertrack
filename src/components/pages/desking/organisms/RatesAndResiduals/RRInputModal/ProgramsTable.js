'use strict';
import React from 'react';
import PropTypes from 'prop-types';
import Radio from 'react-bootstrap/lib/Radio';
import DataTable from '../../../../../reusable/DataTable/DataTable';
import ObjectUtil from '../../../../../../utils/ObjectUtil';
import Program from '../../../../../../models/RatesAndResiduals/Program';

// eslint-disable-next-line react/prefer-stateless-function
class ProgramsTable extends React.PureComponent {
  constructor(props, context) {
    super(props, context);
  }

  selector = (props)=>{
    const onClick = () =>{
      const selectedProgram = ObjectUtil.clone(new Program,props.rowData);
      this.props.setSelectedProgram(selectedProgram);
    };

    const onChange = ()=>{

    };
    
    const checked = !!(this.props.selectedProgram && (this.props.selectedProgram.id === props.rowData.id));
    return (
      <div>
        <Radio
          id={`ProgramRadio${props.rowData.id}`}
          checked={checked}
          onClick={onClick}
          onChange={onChange}/>
      </div>
    );
  }

  render() {
    const columns =[
      {
        name:'',
        header:' ',
        rowComponent: this.selector,
        width:10
      },
      {
        name:'type',
        header:'Type',
        width:200
      },
      {
        name:'lender',
        header:'Lender',
        width:200
      },
      {
        name:'rebates',
        header:'Rebates',
        width:200
      },
      {
        name:'rateMoneyFactor',
        header:'Rate/MF',
        width:200
      },
      {
        name:'term',
        header:'Term',
        width:200
      },
      {
        name:'acquisitionFee',
        header:'Acq Fee',
        width:200
      },
      {
        name:'residual',
        header:'Res %',
        width:200
      },
      {
        name:'milesPerYear',
        header:'Miles/yr',
        width:200
      },
      {
        name:'payment',
        header:'Payment',
        width:200
      },
      {
        name:'frontGross',
        header:'F. Gross',
        width:200
      },
      {
        name:'backGross',
        header:'B. Gross',
        width:200
      },
      {
        name:'expirationDate',
        header:'Exp Date',
        width:200
      },
      {
        name:'maxAdvance',
        header:'Max Adv',
        width:200
      },
      {
        name:'amountFinanced',
        header:'Amnt Financed'
      }
    ];
    return (
      <DataTable
        htmlId="RRProgramsTable"
        data={this.props.programs}
        columns={columns}
      />
    );
  }
}

ProgramsTable.propTypes = {
  setSelectedProgram: PropTypes.func,
  selectedProgram: PropTypes.instanceOf(Program),
  programs: PropTypes.arrayOf(PropTypes.instanceOf(Program))
};

export default ProgramsTable;
