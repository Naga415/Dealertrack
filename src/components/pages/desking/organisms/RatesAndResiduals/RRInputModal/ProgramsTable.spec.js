'use strict';
import React from 'react';
import { mount } from 'enzyme';
import sinon from 'sinon';
import renderer from 'react-test-renderer';
import ProgramsTable from './ProgramsTable';
import Program from '../../../../../../models/RatesAndResiduals/Program';
import * as ScenarioTypes from '../../../../../../models/ScenarioTypes';

const financeProgram = new Program();
financeProgram.id = 0;
financeProgram.type = ScenarioTypes.FINANCE;
financeProgram.acquisitionFee = 456;
financeProgram.amountFinanced = 40000;
financeProgram.backGross = 456;
financeProgram.expirationDate = '11/23/2020';
financeProgram.frontGross = 234;
financeProgram.lender = 'acd';
financeProgram.maxAdvance = 643;
financeProgram.payment = 900;
financeProgram.rebates =1500;
financeProgram.term = 24;
financeProgram.rateMoneyFactor = 6.9;

const balloonProgram = new Program();
balloonProgram.id = 1;
balloonProgram.type = ScenarioTypes.BALLOON;
balloonProgram.acquisitionFee = 456;
balloonProgram.amountFinanced = 40000;
balloonProgram.backGross = 456;
balloonProgram.expirationDate = '11/23/2018';
balloonProgram.frontGross = 234;
balloonProgram.lender = 'acd';
balloonProgram.maxAdvance = 643;
balloonProgram.payment = 480;
balloonProgram.rebates =0;
balloonProgram.term = 60;
balloonProgram.rateMoneyFactor = 6.9;

const leaseProgram = new Program();
leaseProgram.id = 2;
leaseProgram.type = ScenarioTypes.LEASE;
leaseProgram.acquisitionFee = 456;
leaseProgram.amountFinanced = 40000;
leaseProgram.backGross = 456;
leaseProgram.expirationDate = '11/23/2018';
leaseProgram.frontGross = 234;
leaseProgram.lender = 'acd';
leaseProgram.maxAdvance = 643;
leaseProgram.payment = 552;
leaseProgram.rebates =0;
leaseProgram.term = 60;
leaseProgram.rateMoneyFactor = .0035;
leaseProgram.milesPerYear = 10000;
leaseProgram.residual= .45;


const lenders = ['UMB', 'Ally'];
const programs = [financeProgram, balloonProgram, leaseProgram];
const selectedProgram = financeProgram;

const setSelectedProgram = sinon.spy();

function setupProps(lenders, programs, selectedProgram){
  return {
    lenders,
    programs,
    selectedProgram,
    setSelectedProgram
  };
}

function setupmount (lenders, programs, selectedProgram) {
  const props = setupProps(lenders, programs, selectedProgram);
  return mount(<ProgramsTable {...props} />);
}

describe('ProgramsTable', ()=>{
  it('renders <ProgramsTable/> ', () => {
    const props = setupProps(lenders, programs, selectedProgram);
    const component = renderer.create(<ProgramsTable {...props}/>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render <ProgramsTable/> component with 0 programs selected', () => {
    const component = setupmount(lenders, programs);
    expect(component.find({type:'radio',checked:true}).length).toBe(0);
  });

  it('should render <ProgramsTable/> component with 1 selected program', () => {
    const component = setupmount(lenders, programs, selectedProgram);
    expect(component.find({type:'radio',checked:true}).length).toBe(1);
  });

  it('triggers setSelectedProgram event when program radio is clicked', () => {
    const component = setupmount(lenders, programs);
    component.find(`#ProgramRadio${financeProgram.id}`).simulate('click');
    expect(setSelectedProgram.calledOnce).toEqual(true);
  });
});
