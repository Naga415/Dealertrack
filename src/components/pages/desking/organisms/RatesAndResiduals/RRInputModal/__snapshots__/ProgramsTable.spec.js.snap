// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgramsTable renders <ProgramsTable/>  1`] = `
<div
  className="data-table undefined"
  id="RRProgramsTable"
>
  <div
    className={undefined}
    style={undefined}
  >
    <input
      className="griddle-filter form-control"
      name="filter"
      onChange={[Function]}
      placeholder="Filter"
      style={undefined}
      type="text"
    />
    <div
      className={undefined}
      style={undefined}
    />
    <table
      className="griddle-table"
      style={undefined}
    >
      <thead
        className="griddle-table-heading"
        style={undefined}
      >
        <tr>
          <th
            className="griddle-table-heading-cell"
            onClick={[Function]}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={undefined}
          >
            <span>
               
            </span>
          </th>
          <th
            className="griddle-table-heading-cell"
            onClick={[Function]}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={undefined}
          >
            <span>
              Type
            </span>
          </th>
          <th
            className="griddle-table-heading-cell"
            onClick={[Function]}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={undefined}
          >
            <span>
              Lender
            </span>
          </th>
          <th
            className="griddle-table-heading-cell"
            onClick={[Function]}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={undefined}
          >
            <span>
              Rebates
            </span>
          </th>
          <th
            className="griddle-table-heading-cell"
            onClick={[Function]}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={undefined}
          >
            <span>
              Rate/MF
            </span>
          </th>
          <th
            className="griddle-table-heading-cell"
            onClick={[Function]}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={undefined}
          >
            <span>
              Term
            </span>
          </th>
          <th
            className="griddle-table-heading-cell"
            onClick={[Function]}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={undefined}
          >
            <span>
              Acq Fee
            </span>
          </th>
          <th
            className="griddle-table-heading-cell"
            onClick={[Function]}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={undefined}
          >
            <span>
              Res %
            </span>
          </th>
          <th
            className="griddle-table-heading-cell"
            onClick={[Function]}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={undefined}
          >
            <span>
              Miles/yr
            </span>
          </th>
          <th
            className="griddle-table-heading-cell"
            onClick={[Function]}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={undefined}
          >
            <span>
              Payment
            </span>
          </th>
          <th
            className="griddle-table-heading-cell"
            onClick={[Function]}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={undefined}
          >
            <span>
              F. Gross
            </span>
          </th>
          <th
            className="griddle-table-heading-cell"
            onClick={[Function]}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={undefined}
          >
            <span>
              B. Gross
            </span>
          </th>
          <th
            className="griddle-table-heading-cell"
            onClick={[Function]}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={undefined}
          >
            <span>
              Exp Date
            </span>
          </th>
          <th
            className="griddle-table-heading-cell"
            onClick={[Function]}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={undefined}
          >
            <span>
              Max Adv
            </span>
          </th>
          <th
            className="griddle-table-heading-cell"
            onClick={[Function]}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={undefined}
          >
            <span>
              Amnt Financed
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        className="griddle-table-body"
        style={undefined}
      >
        <tr
          className="griddle-row"
          style={undefined}
        >
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 10,
              }
            }
          >
            <div>
              <div
                className="radio"
                style={undefined}
              >
                <label
                  title=""
                >
                  <input
                    checked={true}
                    disabled={false}
                    id="ProgramRadio0"
                    onChange={[Function]}
                    onClick={[Function]}
                    type="radio"
                  />
                </label>
              </div>
            </div>
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            Finance
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            acd
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            1500
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            6.9
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            24
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            456
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          />
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          />
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            900
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            234
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            456
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            11/23/2020
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            643
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": null,
              }
            }
          >
            40000
          </td>
        </tr>
        <tr
          className="griddle-row"
          style={undefined}
        >
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 10,
              }
            }
          >
            <div>
              <div
                className="radio"
                style={undefined}
              >
                <label
                  title=""
                >
                  <input
                    checked={false}
                    disabled={false}
                    id="ProgramRadio1"
                    onChange={[Function]}
                    onClick={[Function]}
                    type="radio"
                  />
                </label>
              </div>
            </div>
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            Balloon
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            acd
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            0
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            6.9
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            60
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            456
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          />
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          />
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            480
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            234
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            456
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            11/23/2018
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            643
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": null,
              }
            }
          >
            40000
          </td>
        </tr>
        <tr
          className="griddle-row"
          style={undefined}
        >
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 10,
              }
            }
          >
            <div>
              <div
                className="radio"
                style={undefined}
              >
                <label
                  title=""
                >
                  <input
                    checked={false}
                    disabled={false}
                    id="ProgramRadio2"
                    onChange={[Function]}
                    onClick={[Function]}
                    type="radio"
                  />
                </label>
              </div>
            </div>
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            Lease
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            acd
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            0
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            0.0035
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            60
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            456
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            0.45
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            10000
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            552
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            234
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            456
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            11/23/2018
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": 200,
              }
            }
          >
            643
          </td>
          <td
            className="griddle-cell"
            onClick={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            style={
              Object {
                "width": null,
              }
            }
          >
            40000
          </td>
        </tr>
      </tbody>
    </table>
    <div
      className="griddle-pagination"
      style={undefined}
    >
      <ul
        className="pagination pagination-md"
      >
        <li
          className="disabled"
          style={undefined}
        >
          <a
            href="#"
            onClick={[Function]}
            role="button"
            style={
              Object {
                "pointerEvents": "none",
              }
            }
            tabIndex={-1}
          >
            <span
              aria-label="First"
            >
              «
            </span>
          </a>
        </li>
        <li
          className="disabled"
          style={undefined}
        >
          <a
            href="#"
            onClick={[Function]}
            role="button"
            style={
              Object {
                "pointerEvents": "none",
              }
            }
            tabIndex={-1}
          >
            <span
              aria-label="Previous"
            >
              ‹
            </span>
          </a>
        </li>
        <li
          className="active"
          style={undefined}
        >
          <a
            href="#"
            onClick={[Function]}
            role="button"
          >
            1
          </a>
        </li>
        <li
          className="disabled"
          style={undefined}
        >
          <a
            href="#"
            onClick={[Function]}
            role="button"
            style={
              Object {
                "pointerEvents": "none",
              }
            }
            tabIndex={-1}
          >
            <span
              aria-label="Next"
            >
              ›
            </span>
          </a>
        </li>
        <li
          className="disabled"
          style={undefined}
        >
          <a
            href="#"
            onClick={[Function]}
            role="button"
            style={
              Object {
                "pointerEvents": "none",
              }
            }
            tabIndex={-1}
          >
            <span
              aria-label="Last"
            >
              »
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
